src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -535,17 +535,43 @@ public static int findNumberOfRepetitions(byte[] repeatUnit, byte[] testString,
         // look backward. For example, if repeatUnit = AT and testString = GATAT, number of repeat units is still 2
         // look forward on the test string
         for (int start = testString.length - repeatUnit.length; start >= 0; start -= repeatUnit.length) {
-            int end = start + repeatUnit.length;
-            byte[] unit = Arrays.copyOfRange(testString,start, end);
-            if(Arrays.equals(unit,repeatUnit))
+            final int end = start + repeatUnit.length;
+            if(equalToSubarray(testString,start, end, repeatUnit)) {
                 numRepeats++;
-            else
+            } else {
                 break;
+            }
         }
         return numRepeats;
     }
 
     /**
+     *  Checks if arr1 starting at 'from' and ending right before 'to' equals to arr2.
+     *  The 'to' argument may lie outside of arr1 and then the 'elements' beyond arr1 are treated as zero.
+     *  In other words this is equivalent to:
+     <code>
+             byte[] unit = Arrays.copyOfRange(arr1, from, to);    //this allows 'to' to be outside of arr1
+             return Arrays.equals(unit,arr2);
+     </code>
+     *  but avoids making a new array instance.
+     */
+    private static boolean equalToSubarray(final byte[] arr1, final int from, final int to, final byte[] arr2){
```

Comment:
This should be in `Utils` rather than embedded within this class, and should (probably) throw if `from` and `to` are out of bounds on `arr1` (unless the calling code actually requires this, which seems unlikely).